File: eval/fixtures/py-sqli/vulnerable.py
# eval/fixtures/py-sqli/vulnerable.py
# BACO Eval Fixture: SQL Injection via f-string (CWE-89)
# The vulnerability is on line 15 - f-string interpolation in SQL query

from typing import Optional

def get_user_by_id(user_id: int) -> Optional[dict]:
    """
    Fetch user from database by ID.
    
    VULNERABLE: Uses f-string interpolation in SQL query.
    An attacker can inject arbitrary SQL via the user_id parameter.
    """
    query = f"SELECT * FROM users WHERE id = {user_id}"
    
    # Execute the query (mock implementation)
    # In production, this would connect to a real database
    print(f"Executing: {query}")
    
    return None


def delete_user(user_id: int) -> bool:
    """
    Delete a user by ID.
    
    VULNERABLE: Same f-string SQL injection pattern.
    """
    query = f"DELETE FROM users WHERE id = {user_id}"
    print(f"Executing: {query}")
    return True

File: eval/fixtures/php-sqli/vulnerable.php
<?php
// eval/fixtures/php-sqli/vulnerable.php
// BACO Eval Fixture: SQL Injection via $_GET concatenation (CWE-89)
// The vulnerability is on line 16 - $_GET value concatenated into SQL

require_once 'db.php';

/**
 * Look up a user by the id request parameter.
 *
 * VULNERABLE: the raw $_GET['id'] value is concatenated into the SQL
 * statement, so an attacker can inject arbitrary SQL (e.g. ' OR 1=1 --).
 */
function find_user($connection) {
    $user_id = $_GET['id'];
    $query = "SELECT * FROM users WHERE id = '" . $user_id . "'";
    $result = mysqli_query($connection, $query);
    return mysqli_fetch_assoc($result);
}
